src/ResourceManager/Consumption/Commands.Consumption.Test/ScenarioTests/BudgetTests.ps1

```diff
@@ -0,0 +1,143 @@
+﻿# ----------------------------------------------------------------------------------
+#
+# Copyright Microsoft Corporation
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+# http://www.apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ----------------------------------------------------------------------------------
+
+<#
+.SYNOPSIS
+Get Budget name
+#>
+function Get-BudgetName
+{
+    return "Budget-" + (getAssetName)
+}
+
+<#
+.SYNOPSIS
+Get Notification key
+#>
+function Get-NotificationKey
+{
+    return "NotificationKey-" + (getAssetName)
+}
+
+<#
+.SYNOPSIS
+Test budget at subscription level
+#>
+function Test-BudgetAtSubscriptionLevel
+{
+	# Setup
+	$budgetName = Get-BudgetName
+	$notificationKey = Get-NotificationKey
+	# This resource group is created at the scope of test subscription
+	$resourceGroupName = "CriTest"
```

Comment:
Please use "getAssetName" here as well, and create the resourcegroup at the beginning and destroy it at the end.  Unless there is special information you need the resource group to have, it seems like there is no reason to hardcode the resource group here.

Reply:
Just saw your comment - if the resourcegroup does not need to exist here, then you can just change this to getAssetName and not create/destroy it.

Reply:
@maddieclayton Will call New-AzureRmResourceGroup to create and later destroy it in testing